go/tools/go-upgrade/go_upgrade.go

```diff
@@ -0,0 +1,498 @@
+
+	versre := regexp.MustCompile("(?i).*BOOTSTRAP_VERSION[[:space:]]*=[[:space:]]*([0-9.]+).*")
+	versionStr := versre.FindStringSubmatch(content)
+	if len(versionStr) != 2 {
+		return 0, fmt.Errorf("malformatted error, got: %v", versionStr)
+	}
+	f, err := strconv.ParseFloat(versionStr[1], 64)
+	if err != nil {
+		return 0, err
+	}
+	return f, nil
+}
+
+// getLatestStableGolangReleases fetches the latest stable releases of Golang from
+// the official website using the goDevAPI URL.
+// Once fetched, the releases are returned as version.Collection.
+func getLatestStableGolangReleases() (version.Collection, error) {
+	resp, err := http.Get(goDevAPI)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	var latestGoReleases []latestGolangRelease
+	err = json.Unmarshal(body, &latestGoReleases)
+	if err != nil {
+		return nil, err
+	}
+
+	var versions version.Collection
+	for _, release := range latestGoReleases {
+		if !release.Stable {
+			continue
+		}
+		if !strings.HasPrefix(release.Version, "go") {
+			return nil, fmt.Errorf("golang version malformatted: %s", release.Version)
+		}
+		newVersion, err := version.NewVersion(release.Version[2:])
+		if err != nil {
+			return nil, err
+		}
+		versions = append(versions, newVersion)
+	}
+	return versions, nil
+}
+
+// chooseNewVersion decides what will be the next version we're going to use in our codebase.
+// Given the current Golang version, the available latest versions and whether we allow major upgrade or not,
+// chooseNewVersion will return either the new version or nil if we cannot/don't need to upgrade.
+func chooseNewVersion(curVersion *version.Version, latestVersions version.Collection, allowMajorUpgrade bool) *version.Version {
+	selectedVersion := curVersion
+	for _, latestVersion := range latestVersions {
+		if !allowMajorUpgrade && !isSameMajorMinorVersion(latestVersion, selectedVersion) {
+			continue
+		}
+		if latestVersion.GreaterThan(selectedVersion) {
+			selectedVersion = latestVersion
+		}
+	}
+	// No change detected, return nil meaning that we do not want to have a new Golang version.
+	if selectedVersion.Equal(curVersion) {
+		return nil
+	}
+	return selectedVersion
+}
+
+// replaceGoVersionInCodebase goes through all the files in the codebase where the
+// Golang version must be updated
+func replaceGoVersionInCodebase(old, new *version.Version, workflowUpdate bool) error {
+	if old.Equal(new) {
+		return nil
+	}
+	explore := []string{
+		"./test/templates",
+		"./build.env",
+		"./docker/bootstrap/Dockerfile.common",
+	}
+	if workflowUpdate {
+		explore = append(explore, "./.github/workflows")
+	}
+	filesToChange, err := getListOfFilesInPaths(explore)
+	if err != nil {
+		return err
+	}
+
+	for _, fileToChange := range filesToChange {
+		err = replaceInFile(
+			[]*regexp.Regexp{regexp.MustCompile(fmt.Sprintf(`(%s)`, old.String()))},
+			[]string{new.String()},
```

Comment:
```suggestion
			[]*regexp.Regexp{
				regexp.MustCompile(`(?i)ARG[[:space:]]*bootstrap_version[[:space:]]*=[[:space:]]*([0-9.]+)`), // Dockerfile
				regexp.MustCompile(`(?i)BOOTSTRAP_VERSION[[:space:]]*=[[:space:]]*[0-9.]+`),                  // Makefile
			},
			[]string{
				fmt.Sprintf("ARG bootstrap_version=%-1g", new), // Dockerfile
				fmt.Sprintf("BOOTSTRAP_VERSION=%-1g", new),     // Makefile
			},
```

Reply:
Fixed it via c6f56d377a too, thanks for catching this :)